src/backersManager/newAllocation.ts

```diff
@@ -0,0 +1,134 @@
+import { DEFAULT_BIGINT } from "../utils";
+import { Address, BigInt } from "@graphprotocol/graph-ts";
+
+
+export function handleNewAllocation(event: NewAllocationEvent): void {
+  _handleBackerStakingHistory(event);
+  _handleBuilder(event);
+  _handleBacker(event);
+  _handleBackerToBuilder(event);
+}
+
+function _handleBackerStakingHistory(event: NewAllocationEvent): void {
+  const backerAddress = event.params.backer_;
+  const backersManagerContract = BackersManagerRootstockCollectiveContract.bind(
+    event.address
+  );
+  const backerTotalAllocation =
+    backersManagerContract.backerTotalAllocation(backerAddress);
+
+  let backer = BackerStakingHistory.load(backerAddress);
+
+  if (backer == null) {
+    backer = new BackerStakingHistory(backerAddress);
+    backer.accumulatedTime = DEFAULT_BIGINT;
+    backer.backerTotalAllocation = DEFAULT_BIGINT;
+  }
+
+  if (backer.backerTotalAllocation.gt(DEFAULT_BIGINT)) {
+    const lastStakedSeconds = event.block.timestamp.minus(
+      backer.lastBlockTimestamp
+    );
+    backer.accumulatedTime = backer.accumulatedTime.plus(lastStakedSeconds);
+  }
+  backer.lastBlockNumber = event.block.number;
+  backer.lastBlockTimestamp = event.block.timestamp;
+  backer.backerTotalAllocation = backerTotalAllocation;
+
+  backer.save();
+
+  const gaugeId = backerAddress.concat(event.params.gauge_);
+  let gauge = GaugeStakingHistory.load(gaugeId);
+  if (gauge == null) {
+    gauge = new GaugeStakingHistory(gaugeId);
+    gauge.gauge = event.params.gauge_;
+    gauge.accumulatedAllocationsTime = DEFAULT_BIGINT;
+    gauge.backer = backerAddress;
+  } else {
+    const lastStakedSeconds = event.block.timestamp.minus(
+      gauge.lastBlockTimestamp
+    );
+    gauge.accumulatedAllocationsTime = gauge.accumulatedAllocationsTime.plus(
+      gauge.allocation.times(lastStakedSeconds)
+    );
+  }
+  gauge.allocation = event.params.allocation_;
+  gauge.lastBlockNumber = event.block.number;
+  gauge.lastBlockTimestamp = event.block.timestamp;
+
+  gauge.save();
+}
+
+function _getPreviousAllocation(gaugeToBuilder: GaugeToBuilder, backer: Address): BigInt {
+  let previousAllocation = DEFAULT_BIGINT;
+  const backerToBuilder = BackerToBuilder.load(
+    backer.concat(gaugeToBuilder.builder)
+  );
+  if (backerToBuilder != null) {
+    previousAllocation = backerToBuilder.totalAllocation;
+  }
+
+  return previousAllocation;
+}
+
+function _handleBuilder(event: NewAllocationEvent): void {
+  const gaugeToBuilder = GaugeToBuilder.load(event.params.gauge_);
+  if (gaugeToBuilder == null) return;
+
+  const builder = Builder.load(gaugeToBuilder.builder);
+  if (builder == null) return;
+
+  const previousAllocation = _getPreviousAllocation(gaugeToBuilder, event.params.backer_);
+  builder.totalAllocation = builder.totalAllocation.minus(previousAllocation).plus(event.params.allocation_);
```

Comment:
I'm not sure that I understand this correctly.
The `_getPreviousAllocation` always returns `builder.totalAllocation` (as the builder has to exist to get here).
And so the `builder.totalAllocation.minus(previousAllocation)` is basically `builder.totalAllocation - builder.totalAllocation` and thus 0. That in turn means that the `builder.totalAllocation = builder.totalAllocation.minus(previousAllocation).plus(event.params.allocation_);` is equivalent to `builder.totalAllocation = event.params.allocation_;`

What am I missing?